Wrappers/Python/ccpi/viewer/widgets/box_widgets.py

```diff
@@ -206,64 +210,72 @@ def GetBoxBoundsFromEventPosition(viewer, position, scale_factor=0.3):
+            else:
+                dialog.exec()
 
-            # Set top right point.
-            # Top right is xmax, zmax
-            box_pos[1] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[0], world_extent, "x")
-            box_pos[5] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[4], world_extent, "z")
-
-            # Set the scroll axis to maximum extent eg. min-max
-            # ymin, ymax
-            box_pos[2] = world_image_min[orientation]
-            box_pos[3] = world_extent[orientation]
-
-        else:
+        elif orientation == SLICE_ORIENTATION_YZ:
             # orientation == 0
             # Looking along x
-            # Lower left is ymin, zmin
-            box_pos[2] = world_mouse_pos[1]
-            box_pos[4] = world_mouse_pos[2]
-
-            # Set top right point
-            # Top right is ymax, zmax
-            box_pos[3] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[2], world_extent, "y")
-            box_pos[5] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[4], world_extent, "z")
-
-            # Set the scroll axis to maximum extent eg. min-max
-            # xmin, xmax
-            box_pos[0] = world_image_min[orientation]
-            box_pos[1] = world_extent[orientation]
+            if 0 <= world_mouse_pos[1] <= world_image_max[1] and 0 <= world_mouse_pos[2] <= world_image_max[2]:
+                # Lower left is ymin, zmin
+                box_pos[2] = world_mouse_pos[1]
+                box_pos[4] = world_mouse_pos[2]
+
+                # Set top right point
+                # Top right is ymax, zmax
+                box_pos[3] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[2], world_extent, "y")
+                box_pos[5] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[4], world_extent, "z")
+
+                # Set the scroll axis to maximum extent eg. min-max
+                # xmin, xmax
+                box_pos[0] = world_image_min[orientation]
+                box_pos[1] = world_image_max[orientation]
+            else:
+                dialog.exec()
+
+        elif orientation == SLICE_ORIENTATION_XZ:
+            # Looking along y
+            if 0 <= world_mouse_pos[2] <= world_image_max[2] and 0 <= world_mouse_pos[0] <= world_image_max[0]:
+                # Lower left is xmin, zmin
+                box_pos[4] = world_mouse_pos[2]
+                box_pos[0] = world_mouse_pos[0]
+
+                # Set top right point.
+                # Top right is xmax, zmax
+                box_pos[5] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[4], world_extent, "z")
+                box_pos[1] = cilviewerBoxWidget.GetTruncatedBoxCoord(box_pos[0], world_extent, "x")
+
+                # Set the scroll axis to maximum extent eg. min-max
+                # ymin, ymax
+                box_pos[2] = world_image_min[orientation]
+                box_pos[3] = world_image_max[orientation]
+            else:
+                dialog.exec()
```

Comment:
Let's open an issue about this code in order to simplify it and use a generic function.

Reply:
Ok, perhaps once we work on https://github.com/TomographicImaging/CILViewer/issues/426 we can tidy this as well. I will add the note in the same issue.